File: HUI2/src/services/CartService.js
import axios from "axios";

const API = `${import.meta.env.VITE_API_URL}/cart`;

export const getCart = (email) => axios.get(`${API}/${email}`);

export const addToCart = (product) => axios.post(API, product);

export const deleteCart = (id) => axios.delete(`${API}/${id}`);

export const updateCartQuantity = (id, quantity) => axios.put(`${API}/${id}`, { quantity });

File: HerbalLife/src/main/java/com/example/HerbalLife/service/CartService.java
package com.example.HerbalLife.service;

import java.util.List;

import com.example.HerbalLife.entity.Cart;

public interface CartService {

    Cart addToCart(Cart cart);

    List<Cart> getCart(String email);

    void deleteCart(String id);

    Cart updateQuantity(String id, int quantity);

}